### DISCUSSION TITLE
Trying to upgrade to 24.1

### DISCUSSION THREAD

**Discussion by thomasbeaudry** [aces/Loris#8228] [General]
Hi,   I'm running CentOS and trying to upgrade to 24.1.  After successfully running the SQL patches, when I run make all I get the following error:

$ make all
npm install
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated core-js@1.2.7: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.

> @fortawesome/fontawesome-free@5.15.4 postinstall /var/www/loris/node_modules/@fortawesome/fontawesome-free
> node attribution.js

Font Awesome Free 5.15.4 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)

> loris@1.0.0 postinstall /var/www/loris
> cd modules/electrophysiology_browser/jsx/react-series-data-viewer && npm install

audited 481 packages in 5.265s
found 138 vulnerabilities (6 low, 36 moderate, 85 high, 11 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN The package style-loader is included as both a dev and production dependency.

added 1471 packages from 704 contributors and audited 1474 packages in 57.023s
found 10 vulnerabilities (1 low, 2 moderate, 7 high)
  run `npm audit fix` to fix them, or `npm audit` for details
npm run compile

> loris@1.0.0 compile /var/www/loris
> webpack

Child

    WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
    This can impact web performance.
    Assets:
      ../../vendor/js/react/react-dom.development.js (918 KiB)

    WARNING in webpack performance recommendations:
    You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
    For more info visit https://webpack.js.org/guides/code-splitting/

    ERROR in .eslintrc.json:
        Configuration for rule "extends" is invalid:
        Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"eslint:recommended"').

Child

    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx 165:7-15
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx(165,8)
          TS2786: 'Provider' cannot be used as a JSX component.
      Its instance type 'Provider<AnyAction>' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/node_modules/@types/hoist-non-react-statics/node_modules/@types/react/index").ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is not assignable to type 'ReactPortal'.

    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 98:11-15
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(98,12)
          TS2786: 'Rail' cannot be used as a JSX component.
      Its instance type 'Rail' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'string | number | boolean | {} | ReactNodeArray | ReactPortal | ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<...>)>' is not assignable to type 'ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is missing the following properties from type 'ReactPortal': key, children, type, props

    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 103:11-18
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(103,12)
          TS2786: 'Handles' cannot be used as a JSX component.
      Its instance type 'Handles' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'string | number | boolean | {} | ReactNodeArray | ReactPortal | ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<...>)>' is not assignable to type 'ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is not assignable to type 'ReactPortal'.

    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 117:11-16
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(117,12)
          TS2786: 'Ticks' cannot be used as a JSX component.
      Its instance type 'Ticks' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<any, any, any>)>' is not assignable to type 'ReactNode'.
            Property 'children' is missing in type 'ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<any, any, any>)>' but required in type 'ReactPortal'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! loris@1.0.0 compile: `webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the loris@1.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-11-09T19_59_54_303Z-debug.log
make: *** [javascript] Error 2

Thanks for your help!

**Comment by ridz1208** [comment]
@laemtl help

**Comment by ridz1208** [comment]
@thomasbeaudry Hey, I have gotten more info thanks to @laemtl. the bug you are getting has already been fixed on the `24.1-release` and will very soon be released to v24.1.1 bugfix release.

That being said, once you pull the new code in make sure to run the `make` command (you might need to update your node and npm versions to the latest STABLE versions available before running make)

personally im running the following versions
```
$ node -v
v16.18.0
$ npm -v
8.19.2
```

**Reply by driusan** [reply]
Hi Tom, I think this was already fixed but we hadn't done a bugfix release for it. Can you try v24.1.1?

**Comment by thomasbeaudry** [comment]
@driusan I tried it again with v24.1.1 but I get a different error now (see below).  @ridz1208 I'm having problems upgrading node and npm, so I don't have the latest versions like you

Error:
$ make all
npm install

> loris@1.0.0 postinstall /var/www/loris
> cd modules/electrophysiology_browser/jsx/react-series-data-viewer && npm install

npm WARN react-redux@8.0.5 requires a peer of react-native@>=0.59 but none is installed. You must install peer dependencies yourself.

audited 480 packages in 8.122s
found 138 vulnerabilities (6 low, 36 moderate, 85 high, 11 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
npm WARN The package style-loader is included as both a dev and production dependency.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 1474 packages in 19.571s
found 10 vulnerabilities (1 low, 2 moderate, 7 high)
  run `npm audit fix` to fix them, or `npm audit` for details
npm run compile

> loris@1.0.0 compile /var/www/loris
> webpack

Child

    WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
    This can impact web performance.
    Assets:
      ../../vendor/js/react/react-dom.development.js (918 KiB)

    WARNING in webpack performance recommendations:
    You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
    For more info visit https://webpack.js.org/guides/code-splitting/

    ERROR in .eslintrc.json:
        Configuration for rule "extends" is invalid:
        Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"eslint:recommended"').

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! loris@1.0.0 compile: `webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the loris@1.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-11-10T19_38_09_892Z-debug.log
make: *** [javascript] Error 2

**Reply by driusan** [reply]
@maltheism @laemtl have either of you seen that error before? Seems to be something related to npm.

**Reply by thomasbeaudry** [reply]
Okay so i upgraded node and npm, and i get the same error, i'm running:

$ npm -v
7.21.1
$ node -v
v16.9.1

**Comment by maltheism** [comment]
Hi @thomasbeaudry, I think the `ERROR in .eslintrc.json` might be indicating you have an older installation of ESLint that's overriding the needed version that Loris uses and thus resulting in the error you're having. Try the command `npm list -g --depth=0` to list your global npm packages. Then try the command `npm list` while inside your LORIS directory to see the npm packages of your LORIS installation. We can then compare the version you have to package.json.

**Reply by thomasbeaudry** [reply]
$ npm list -g --depth=0
/usr/local/lib
├── corepack@0.9.0
└── npm@7.21.1

(loris-mri-python) [lorisadmin@loris loris]$ npm list
loris@1.0.0 /var/www/loris
├── @babel/cli@7.19.3
├── @babel/core@7.20.2
├── @babel/plugin-proposal-object-rest-spread@7.20.2
├── @babel/plugin-transform-runtime@7.19.6
├── @babel/preset-env@7.20.2
├── @babel/preset-react@7.18.6
├── @babel/runtime@7.20.1
├── @fortawesome/fontawesome-free@5.15.4
├── @types/react-dom@16.9.17
├── @types/react-redux@7.1.16
├── @types/react@16.14.34
├── alex@11.0.0
├── babel-eslint@10.1.0
├── babel-loader@8.3.0
├── c3@0.7.20
├── copy-webpack-plugin@6.4.1
├── css-loader@3.6.0
├── d3@5.16.0
├── eslint-config-google@0.9.1
├── eslint-plugin-no-jquery@2.7.0
├── eslint-plugin-react@7.31.10
├── eslint-webpack-plugin@2.1.0
├── eslint@7.32.0
├── jstat@1.9.5
├── null-loader@4.0.1
├── papaparse@5.3.2
├── prop-types@15.8.1
├── react-addons-create-fragment@15.6.2
├── react-dom@17.0.2
├── react-markdown@5.0.3
├── react@17.0.2
├── style-loader@1.3.0
├── swagger-ui-dist@3.52.5
├── sweetalert2@8.19.0
├── terser-webpack-plugin@1.4.5
├── ts-loader@8.4.0
├── typescript@3.9.10
├── webpack-cli@3.3.12
└── webpack@4.46.0

**Comment by laemtl** [comment]
Hello @thomasbeaudry, can you share your `package-lock.json` located at the root of Loris? And I'm also curious to see the output of `npm list -g` and /home/lorisadmin/.npm/_logs/2022-11-10T19_38_09_892Z-debug.log.

**Reply by thomasbeaudry** [reply]
$ npm list -g.
loris@1.0.0 /var/www/loris
├── @babel/cli@7.19.3
├── @babel/core@7.20.2
├── @babel/plugin-proposal-object-rest-spread@7.20.2
├── @babel/plugin-transform-runtime@7.19.6
├── @babel/preset-env@7.20.2
├── @babel/preset-react@7.18.6
├── @babel/runtime@7.20.1
├── @fortawesome/fontawesome-free@5.15.4
├── @types/react-dom@16.9.17
├── @types/react-redux@7.1.16
├── @types/react@16.14.34
├── alex@11.0.0
├── babel-eslint@10.1.0
├── babel-loader@8.3.0
├── c3@0.7.20
├── copy-webpack-plugin@6.4.1
├── css-loader@3.6.0
├── d3@5.16.0
├── eslint-config-google@0.9.1
├── eslint-plugin-no-jquery@2.7.0
├── eslint-plugin-react@7.31.10
├── eslint-webpack-plugin@2.1.0
├── eslint@7.32.0
├── jstat@1.9.5
├── null-loader@4.0.1
├── papaparse@5.3.2
├── prop-types@15.8.1
├── react-addons-create-fragment@15.6.2
├── react-dom@17.0.2
├── react-markdown@5.0.3
├── react@17.0.2
├── style-loader@1.3.0
├── swagger-ui-dist@3.52.5
├── sweetalert2@8.19.0
├── terser-webpack-plugin@1.4.5
├── ts-loader@8.4.0
├── typescript@3.9.10
├── webpack-cli@3.3.12
└── webpack@4.46.0

**Reply by laemtl** [reply]
Thanks for your feedback! Unfortunately, the wrong file has been shared. I would like to take a look at package-lock.json, which stores the exact version of your npm packages.

**Reply by thomasbeaudry** [reply]
@laemtl Sorry about that!
[package-lock.zip](https://github.com/aces/Loris/files/9991529/package-lock.zip)

**Reply by thomasbeaudry** [reply]
It seems like my answer disappeared (or a github issue).   I tried make clean && make  as well as make clean && make all afterwards and still got the same error

**Reply by laemtl** [reply]
Can you share the file `/home/lorisadmin/.npm/_logs/2022-11-10T19_38_09_892Z-debug.log` and `modules/electrophysiology_browser/jsx/react-series-data-viewer/package-lock.json`?

**Reply by thomasbeaudry** [reply]
$ cat /home/lorisadmin/.npm/_logs/2022-11-10T19_38_09_892Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'compile' ]
2 info using npm@6.9.0
3 info using node@v10.16.2
4 verbose run-script [ 'precompile', 'compile', 'postcompile' ]
5 info lifecycle loris@1.0.0~precompile: loris@1.0.0
6 info lifecycle loris@1.0.0~compile: loris@1.0.0
7 verbose lifecycle loris@1.0.0~compile: unsafe-perm in lifecycle true
8 verbose lifecycle loris@1.0.0~compile: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/loris/node_modules/.bin:/home/lorisadmin/scripts:/home/lorisadmin/perl5/bin:/data/loris/bin/mri/python_virtualenvs/loris-mri-python/bin:/data/loris/bin/mri:/data/loris/bin/mri/uploadNeuroDB:/data/loris/bin/mri/uploadNeuroDB/bin:/data/loris/bin/mri/dicom-archive:/data/loris/bin/mri/python:/data/loris/bin/mri/tools:/data/loris/bin/mri/python/react-series-data-viewer:/opt/minc/1.9.15//bin:/opt/minc/1.9.15/bin:/opt/minc/1.9.15/pipeline:/opt/minc/1.9.15/bin:/opt/minc/1.9.15/pipeline:/opt/minc/1.9.15/bin:/opt/dcmtk-3.6.0/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/lorisadmin/.local/bin:/home/lorisadmin/bin
9 verbose lifecycle loris@1.0.0~compile: CWD: /var/www/loris
10 silly lifecycle loris@1.0.0~compile: Args: [ '-c', 'webpack' ]
11 silly lifecycle loris@1.0.0~compile: Returned: code: 2  signal: null
12 info lifecycle loris@1.0.0~compile: Failed to exec compile script
13 verbose stack Error: loris@1.0.0 compile: `webpack`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:198:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:198:13)
13 verbose stack     at maybeClose (internal/child_process.js:982:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid loris@1.0.0
15 verbose cwd /var/www/loris
16 verbose Linux 4.1.12-124.60.1.el7uek.x86_64
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "compile"
18 verbose node v10.16.2
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 2
22 error loris@1.0.0 compile: `webpack`
22 error Exit status 2
23 error Failed at the loris@1.0.0 compile script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]

**Comment by maltheism** [comment]
Hi @thomasbeaudry, try the following commands in order:

` npm cache clean --force`
`sudo npm install npm -g`
`rm -rf node_modules`
`rm -f package-lock.json`
`rm -rf modules/electrophysiology_browser/jsx/react-series-data-viewer/node_modules`
`rm -f modules/electrophysiology_browser/jsx/react-series-data-viewer/package-lock.json`
`npm install`
`npm run compile`

Also are you on a clean install of v21.1.1 by using the following commands:
`wget https://github.com/aces/Loris/archive/v24.1.1.tar.gz`
`tar -zxf v24.1.1.tar.gz`
`cp -r v24.1.1/ /var/www/loris`
`chown -R lorisadmin:apache /var/www/loris`

**Comment by thomasbeaudry** [comment]
Hi, I tried all of your commands on a clean install.  I still get:

$ npm run compile

> loris@1.0.0 compile
> webpack

Child

    WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
    This can impact web performance.
    Assets:
      ../../vendor/js/react/react-dom.development.js (918 KiB)

    WARNING in webpack performance recommendations:
    You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
    For more info visit https://webpack.js.org/guides/code-splitting/

    ERROR in .eslintrc.json:
        Configuration for rule "extends" is invalid:
        Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"eslint:recommended"').

**Comment by laemtl** [comment]
Do you have eslint installed globally? Can you try to run `eslint --version`?

**Reply by thomasbeaudry** [reply]
No i don't i get command not found.  Should i install it?

**Reply by laemtl** [reply]
No, having it installed globally could have been one potential source of problem.

**Reply by maltheism** [reply]
Hi @thomasbeaudry @laemtl I'm somewhat confident the issue is an older installation of nodejs & npm are overriding your most latest installed versions.

You made a comment showing:
````
$ npm -v
7.21.1
$ node -v
v16.9.1
````

but the comment of your log show the following:
````
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'compile' ]
2 info using npm@6.9.0
3 info using node@v10.16.2
````

Both npm@6.9.0 and node@v10.16.2 are outdated and would be why the error is happening.

**Reply by thomasbeaudry** [reply]
@maltheism   I just checked again:

(loris-mri-python) [lorisadmin@loris loris]$ npm -v
7.21.1
(loris-mri-python) [lorisadmin@loris loris]$ node -v
v16.9.1

So you think it's a path issue?  I'm not sure where I would set the node/npm paths for LORIS to use since bash is finding the correct one

**Reply by maltheism** [reply]
@thomasbeaudry the issue I believe is that you have two installations of both nodejs & npm on your server. The older installation is being used when you try to build Loris. You can run `/usr/local/bin/node --version` and `/usr/local/bin/npm --version` to see if they exist.

**Reply by thomasbeaudry** [reply]
(loris-mri-python) [lorisadmin@loris loris]$ /usr/local/bin/node --version
v16.9.1
(loris-mri-python) [lorisadmin@loris loris]$ /usr/local/bin/npm --version
7.21.1

**Reply by maltheism** [reply]
@thomasbeaudry that's strange that you're not getting the same versions that are listed from your earlier comment found here: https://github.com/aces/Loris/discussions/8228#discussioncomment-4119199
````
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'compile' ]
2 info using npm@6.9.0
3 info using node@v10.16.2
````

Maybe share the newest log file created when trying to build Loris? It shouldn't show node@v10.16.2 or npm@6.9.0

**Comment by laemtl** [comment]
I think the log with the old node/npm version number is from a run before those got updated.
@thomasbeaudry Can you edit your webpack.config.js and add this line ?

```
     devtool: 'source-map',
     plugins: [
       new ESLintPlugin({
+       eslintPath: 'node_modules/eslint/bin/eslint',
         files: [
           'modules/',
```
Your error indicates that an old version of eslint is running on your system. Since the one installed locally is correct, this may fix your issue.

**Comment by thomasbeaudry** [comment]
@laemtl   I added the line you suggested and now I get a different error at least!

> loris@1.0.0 compile
> webpack

Child

    WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
    This can impact web performance.
    Assets:
      ../../vendor/js/react/react-dom.development.js (918 KiB)

    WARNING in webpack performance recommendations:
    You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
    For more info visit https://webpack.js.org/guides/code-splitting/

    ERROR in Cannot find module 'node_modules/eslint/.bin/eslint'
    Require stack:
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/getESLint.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/linter.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/index.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/cjs.js
    - /var/www/loris/webpack.config.js
    - /var/www/loris/node_modules/webpack-cli/bin/utils/convert-argv.js
    - /var/www/loris/node_modules/webpack-cli/bin/cli.js
    - /var/www/loris/node_modules/webpack/bin/webpack.js
make: *** [javascript] Error 2



I checked the node modules directory and I don't have that file.  But i have:

/var/www/loris/node_modules/eslint/bin/eslint.js

**Reply by thomasbeaudry** [reply]
and if i change the webpack .config.js to point to the eslint.js file, i get this error:


    ERROR in Cannot find module 'node_modules/eslint/bin/eslint.js'
    Require stack:
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/getESLint.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/linter.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/index.js
    - /var/www/loris/node_modules/eslint-webpack-plugin/dist/cjs.js
    - /var/www/loris/webpack.config.js
    - /var/www/loris/node_modules/webpack-cli/bin/utils/convert-argv.js
    - /var/www/loris/node_modules/webpack-cli/bin/cli.js
    - /var/www/loris/node_modules/webpack/bin/webpack.js
make: *** [javascript] Error 2

**Comment by thomasbeaudry** [comment]
@maltheism  My latest build file (I think you mean my npm log ) is this:

$ cat /home/lorisadmin/.npm/_logs/2022-11-10T20_37_21_358Z-debug.log
0 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'update', '-g' ]
1 info using npm@7.21.1
2 info using node@v16.9.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 7ms
6 timing config:load:builtin Completed in 8ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 0ms
9 timing config:load:project Completed in 2ms
10 timing config:load:file:/home/lorisadmin/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 1ms
14 timing config:load:validate Completed in 0ms
15 timing config:load:credentials Completed in 2ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 20ms
18 timing npm:load:configload Completed in 20ms
19 timing npm:load:setTitle Completed in 0ms
20 timing npm:load:setupLog Completed in 1ms
21 timing config:load:flatten Completed in 5ms
22 timing npm:load:cleanupLog Completed in 3ms
23 timing npm:load:configScope Completed in 0ms
24 timing npm:load:projectScope Completed in 1ms
25 timing npm:load Completed in 33ms
26 timing arborist:ctor Completed in 1ms
27 timing idealTree:init Completed in 18ms
28 timing idealTree:userRequests Completed in 4ms
29 silly idealTree buildDeps
30 silly fetch manifest corepack@*
31 http fetch GET 200 https://registry.npmjs.org/corepack 2164ms (cache miss)
32 silly fetch manifest npm@*
33 http fetch GET 200 https://registry.npmjs.org/npm 16ms (cache hit)
34 silly placeDep ROOT corepack@0.15.1 OK for:  want: *
35 silly placeDep ROOT npm@9.1.1 OK for:  want: *
36 timing idealTree:#root Completed in 2216ms
37 timing idealTree:node_modules/corepack Completed in 0ms
38 timing idealTree:node_modules/npm Completed in 0ms
39 timing idealTree:buildDeps Completed in 2218ms
40 timing idealTree:fixDepFlags Completed in 1ms
41 warn EBADENGINE Unsupported engine {
41 warn EBADENGINE   package: 'npm@9.1.1',
41 warn EBADENGINE   required: { node: '^14.17.0 || ^16.13.0 || >=18.0.0' },
41 warn EBADENGINE   current: { node: 'v16.9.1', npm: '7.21.1' }
41 warn EBADENGINE }
42 timing idealTree Completed in 2245ms
43 timing reify:loadTrees Completed in 2620ms
44 timing reify:diffTrees Completed in 4ms
45 silly reify mark retired [
45 silly reify   '/usr/local/lib/node_modules/corepack',
45 silly reify   '/usr/local/bin/corepack',
45 silly reify   '/usr/local/bin/pnpm',
45 silly reify   '/usr/local/bin/pnpx',
45 silly reify   '/usr/local/bin/yarn',
45 silly reify   '/usr/local/bin/yarnpkg'
45 silly reify ]
46 silly reify mark retired [
46 silly reify   '/usr/local/lib/node_modules/npm',
46 silly reify   '/usr/local/bin/npm',
46 silly reify   '/usr/local/bin/npx'
46 silly reify ]
47 silly reify moves {
47 silly reify   '/usr/local/lib/node_modules/corepack': '/usr/local/lib/node_modules/.corepack-xqcTrkeV',
47 silly reify   '/usr/local/bin/corepack': '/usr/local/bin/.corepack-0KY6b4OJ',
47 silly reify   '/usr/local/bin/pnpm': '/usr/local/bin/.pnpm-dB7q63VH',
47 silly reify   '/usr/local/bin/pnpx': '/usr/local/bin/.pnpx-ODwt4u1T',
47 silly reify   '/usr/local/bin/yarn': '/usr/local/bin/.yarn-PedZXo5L',
47 silly reify   '/usr/local/bin/yarnpkg': '/usr/local/bin/.yarnpkg-cxSvgbtB',
47 silly reify   '/usr/local/lib/node_modules/npm': '/usr/local/lib/node_modules/.npm-i9nnxROI',
47 silly reify   '/usr/local/bin/npm': '/usr/local/bin/.npm-uJtxIR1m',
47 silly reify   '/usr/local/bin/npx': '/usr/local/bin/.npx-xXo12k2C'
47 silly reify }
48 timing reify:rollback:retireShallow Completed in 4ms
49 timing command:update Completed in 2710ms
50 notice New major version of npm available! 7.21.1 -> 9.1.1
50 notice Changelog: https://github.com/npm/cli/releases/tag/v9.1.1
50 notice Run npm install -g npm@9.1.1 to update!
51 verbose stack Error: EACCES: permission denied, rename '/usr/local/lib/node_modules/corepack' -> '/usr/local/lib/node_modules/.corepack-xqcTrkeV'
52 verbose cwd /home/lorisadmin
53 verbose Linux 4.1.12-124.60.1.el7uek.x86_64
54 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "update" "-g"
**55 verbose node v16.9.1
56 verbose npm  v7.21.1**
57 error code EACCES
58 error syscall rename
59 error path /usr/local/lib/node_modules/corepack
60 error dest /usr/local/lib/node_modules/.corepack-xqcTrkeV
61 error errno -13
62 error Error: EACCES: permission denied, rename '/usr/local/lib/node_modules/corepack' -> '/usr/local/lib/node_modules/.corepack-xqcTrkeV'
62 error  [Error: EACCES: permission denied, rename '/usr/local/lib/node_modules/corepack' -> '/usr/local/lib/node_modules/.corepack-xqcTrkeV'] {
62 error   errno: -13,
62 error   code: 'EACCES',
62 error   syscall: 'rename',
62 error   path: '/usr/local/lib/node_modules/corepack',
62 error   dest: '/usr/local/lib/node_modules/.corepack-xqcTrkeV'
62 error }
63 error The operation was rejected by your operating system.
63 error It is likely you do not have the permissions to access this file as the current user
63 error
63 error If you believe this might be a permissions issue, please double-check the
63 error permissions of the file and its containing directories, or try running
63 error the command again as root/Administrator.
64 verbose exit -13

**Comment by laemtl** [comment]
@thomasbeaudry Can we have access temporarily to your machine, or a replicate?

**Reply by thomasbeaudry** [reply]
Hi, this issue has been solved by @laemtl THANKS!  I had some old files in my /var/www/loris folder that were interfering when make was compiling.  I'm going to start using git to checkout my releases now, instead of using the official downloaded .tar.gz files on the release page.
